packages/@aws-cdk/aws-servicecatalog/lib/constraints.ts

```diff
@@ -0,0 +1,329 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import * as core from '@aws-cdk/core';
+import * as constructs from 'constructs';
+import { Language } from './language';
+import { IPortfolio } from './portfolio';
+import { IProduct } from './product';
+import * as servicecatalog from './servicecatalog.generated';
+
+/**
+ * Represents an AWS ServiceCatalog constraint
+ */
+export interface IConstraint extends core.IResource {
+  /**
+   * The language code.
+   *
+   * @default en
+   */
+  readonly acceptLanguage?: Language;
+
+  /**
+   * The constraint's description.
+   *
+   * @default none
+   */
+  readonly description?: string;
+
+  /**
+   * The portfolio to associate this constraint to.
+   *
+   * @default none
+   */
+  readonly portfolio?: IPortfolio;
+
+  /**
+   * The product to associate this constraint to.
+   *
+   * @default none
+   */
+  readonly product?: IProduct;
+
+  /**
+   * Applies a constraint to a product/portfolio
+   *
+   * @param product The product to apply the constraint to.
+   * @param portfolio The portfolio containing the product.
+   */
+  apply(product: IProduct, portfolio: IPortfolio): void
+}
+
+interface BaseConstraintProps {
+  /**
+   * The language code.
+   *
+   * @default en
+   */
+  readonly acceptLanguage?: Language;
+  /**
+   * Constraint description
+   *
+   * @default none
+   */
+  readonly description?: string;
+  /**
+   * Portfolio to associate the constraint with.
+   *
+   * @default none
+   */
+  readonly portfolio?: IPortfolio;
+  /**
+   * Product within the portfolio to associate the constraint with.
+   *
+   * @default none
+   */
+  readonly product?: IProduct;
+}
+
+/**
+ * Launch Notification Constraint properties
+ */
+export interface LaunchNotificationConstraintProps extends BaseConstraintProps {
+  /**
+   * A list of topics to notify
+   */
+  readonly topics: sns.ITopic[];
+}
+/**
+ * Launch Role Constraint properties
+ */
+export interface LaunchRoleConstraintProps extends BaseConstraintProps {
+  /**
+   * Role used to launch the stack.
+   */
+  readonly role: iam.IRole;
+}
+/**
+ * Launch Template Constraint properties
+ */
+export interface LaunchTemplateConstraintProps extends BaseConstraintProps {
+  // TODO: make this it's own type instead of a string
+  /**
+   * Launch Template Constraint Rules
+   */
+  readonly rules: string;
```

Comment:
I wasn't able to find much information on how this works. The CfnResource `CfnLaunchTemplateConstraint` expects a string here.